### PR TITLE
Add new Java SDK

### PR DESCRIPTION
**Pull request by arielshaqed**
* In this PR:
  * [x] Also publish new client
  * [x] Test both presigned and "regular" modes
* [ ] Publish this client
* In future PRs:
  * [ ] Use published client in lakeFSFS build, _or_ locally-installed build
  * [x] Add remaining tests

### CODE REVIEW

**Comment by arielshaqed** [comment]
Also fixes #6581 by adding a minimal test that the new client SDK is cool when it sees an unknown property.  This does not really test lakeFS, it tests that OpenAPI-codegen does what we wanted it to do.

**Comment by Jonathan-Rosenberg** [comment]
@arielshaqed can you please specify the files that you actually changed? 
or the ones that you didn't 😉

**Review by arielshaqed — COMMENTED** [review]
@Jonathan-Rosenberg PTAL :-)

I now:
- Moved lakeFSFS changes to a separate PR
- Told Linguist to fold java-legacy by default (see .gitattributes change!)
- Rearranged some changes to be more logical by commit.

It should be easier to review.

File: api/java-gen-ignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Files to ignore (skip) when generating Java client
+maven.yml
+.github/
```
Comment (Jonathan-Rosenberg) [inline api/java-gen-ignore:3]:
```suggestion
.github/
.gitignore
git_push.sh
```
?

Comment (arielshaqed) [reply api/java-gen-ignore:3]: Done, thanks!

Comment (arielshaqed) [reply api/java-gen-ignore:3]: Now really re-done.

**Comment by arielshaqed** [comment]
I think that I added the desired openapi ignore key(?).  PTAL...

**Comment by arielshaqed** [comment]
PTAAAAAAAAAAAAL.

81dqs4

**Review by Jonathan-Rosenberg — APPROVED** [review]
LGTM

**Comment by arielshaqed** [comment]
Thanks!

**Pull request merged**